src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/RuntimeInformation.Windows.cs

```diff
@@ -70,5 +89,28 @@ private static Architecture Map(int processorArchitecture)
                     return Architecture.X86;
             }
         }
+
+        private static Architecture MapMachineConstant(ushort processMachine)
+        {
+            switch (processMachine)
+            {
+                case 0x01C0: // IMAGE_FILE_MACHINE_ARM
+                case 0x01C2: // IMAGE_FILE_MACHINE_THUMB
+                case 0x01C4: // IMAGE_FILE_MACHINE_ARMNT
+                    return Architecture.Arm;
+
+                case 0x8664: // IMAGE_FILE_MACHINE_AMD64
+                    return Architecture.X64;
+
+                case 0xAA64: // IMAGE_FILE_MACHINE_ARM64
+                case 0xA641: // IMAGE_FILE_MACHINE_ARM64EC
+                case 0xA64E: // IMAGE_FILE_MACHINE_ARM64X
```

Comment:
Added these two based on https://github.com/dotnet/diagnostics/pull/2858 (I forgot about looking at that PR).
These weren't in my local winnt.h install, they are defined in Windows 11 SDK: https://blogs.windows.com/windowsdeveloper/2021/06/28/announcing-arm64ec-building-native-and-interoperable-apps-for-windows-11-on-arm/

@jkotas, these are yet another special cases of emulation. Do we need to handle "Emulation Compatible" mode differently?

Reply:
I do not think that `IMAGE_FILE_MACHINE_ARM64EC` and `IMAGE_FILE_MACHINE_ARM64X` can ever be returned by the `IsWow64Process2` as the native host architecture. They do not represent native architectures. And if they were returned, it is unclear what this API should return for those cases. I would delete them from the switch statement.